moveit_ros/moveit_servo/include/moveit_servo/servo_parameters.cpp

```diff
@@ -0,0 +1,278 @@
+    return false;
+  }
+  if (parameters->command_out_type != "trajectory_msgs/JointTrajectory" &&
+      parameters->command_out_type != "std_msgs/Float64MultiArray")
+  {
+    RCLCPP_WARN(logger, "Parameter command_out_type should be "
+                        "'trajectory_msgs/JointTrajectory' or "
+                        "'std_msgs/Float64MultiArray'. Check yaml file.");
+    return false;
+  }
+  if (!parameters->publish_joint_positions && !parameters->publish_joint_velocities &&
+      !parameters->publish_joint_accelerations)
+  {
+    RCLCPP_WARN(logger, "At least one of publish_joint_positions / "
+                        "publish_joint_velocities / "
+                        "publish_joint_accelerations must be true. Check "
+                        "yaml file.");
+    return false;
+  }
+  if ((parameters->command_out_type == "std_msgs/Float64MultiArray") && parameters->publish_joint_positions &&
+      parameters->publish_joint_velocities)
+  {
+    RCLCPP_WARN(logger, "When publishing a std_msgs/Float64MultiArray, "
+                        "you must select positions OR velocities.");
+    return false;
+  }
+  // Collision checking
+  if (parameters->collision_check_type != "threshold_distance" && parameters->collision_check_type != "stop_distance")
+  {
+    RCLCPP_WARN(logger, "collision_check_type must be 'threshold_distance' or 'stop_distance'");
+    return false;
+  }
+  if (parameters->self_collision_proximity_threshold < 0.)
+  {
+    RCLCPP_WARN(logger, "Parameter 'self_collision_proximity_threshold' should be "
+                        "greater than zero. Check yaml file.");
+    return false;
+  }
+  if (parameters->scene_collision_proximity_threshold < 0.)
+  {
+    RCLCPP_WARN(logger, "Parameter 'scene_collision_proximity_threshold' should be "
+                        "greater than zero. Check yaml file.");
+    return false;
+  }
+  if (parameters->scene_collision_proximity_threshold < parameters->self_collision_proximity_threshold)
+  {
+    RCLCPP_WARN(logger, "Parameter 'self_collision_proximity_threshold' should probably be less "
+                        "than or equal to 'scene_collision_proximity_threshold'. Check yaml file.");
+  }
+  if (parameters->collision_check_rate < 0)
+  {
+    RCLCPP_WARN(logger, "Parameter 'collision_check_rate' should be "
+                        "greater than zero. Check yaml file.");
+    return false;
+  }
+  if (parameters->collision_distance_safety_factor < 1)
+  {
+    RCLCPP_WARN(logger, "Parameter 'collision_distance_safety_factor' should be "
+                        "greater than or equal to 1. Check yaml file.");
+    return false;
+  }
+  if (parameters->min_allowable_collision_distance < 0)
+  {
+    RCLCPP_WARN(logger, "Parameter 'min_allowable_collision_distance' should be "
+                        "greater than zero. Check yaml file.");
+    return false;
+  }
+
+  return true;
+}
+
+}  // namespace moveit_servo
```

Comment:
Newline at end of file.  There should be some way to configure your editor to make sure there is a newline at the end of your file when you save it then this annoying issue won't bug you in the future.

Reply:
Indeed there was, fixed it for all files!

Reply:
:+1: